v3/src/data-interactive/handlers/log-message-handler.ts

```diff
@@ -1,8 +1,28 @@
+import { LoggableObject, LoggableValue, logMessageWithReplacement } from "../../lib/log-message"
 import { registerDIHandler } from "../data-interactive-handler"
-import { DIHandler, diNotImplementedYet } from "../data-interactive-types"
+import { DIHandler, DILogMessage, DIResources, DIValues,  } from "../data-interactive-types"
 
 export const diLogMessageHandler: DIHandler = {
-  notify: diNotImplementedYet
+  notify(resources: DIResources, values?: DIValues) {
+    const webviewModel = resources.interactiveFrame?.content
+    if (values) {
+      const {formatStr, replaceArgs} = values as DILogMessage
+      const message = formatStr ?? ""
+      const argsArray = replaceArgs != null
+        ? Array.isArray(replaceArgs) ? replaceArgs : [replaceArgs]
+        : undefined
+      const args: LoggableObject = {}
+      argsArray?.forEach((value: LoggableValue, index: number) => {
+        args[index.toString()] = value
+      })
+      webviewModel?.applyModelChange(()=>{},
+        { log: logMessageWithReplacement(message, args) }
+      )
+      return {success: true}
+    } else {
+      return {success: false, error: "No values provided"}
```

Comment:
If we want to keep this result (see last comment), it should be a translatable string.

Reply:
> If we want to keep this result (see last comment), it should be a translatable string.

I'm not sure that's right. Where logs and the plugin API are concerned, these strings are generally being handled by code (i.e. scripts or plugins) where consistency is likely to be more useful than localizability. That said, if we imagine a particular string being provided for the specific purpose of showing to the user, then maybe a localizable string would make sense, but I'm not sure we have a way of indicating that distinction in the API.

Reply:
This line is an API request result, not a logged message, so I think it makes sense to make it translatable. At least, that's what I've done for all other request messages.

Reply:
👍 Consistency is a virtue at this point.